packages/suite-native-navigation-root/src/dummyScreens/DemoScreen.tsx

```diff
@@ -79,24 +85,28 @@ export const DemoScreen = () => {
                         onChange={() => setIsSwitch2Active(!isSwitch2Active)}
                         isDisabled
                     />
-                    <Box>
-                        <Text variant="titleSmall">Title Small</Text>
-                    </Box>
-                    <Box>
-                        <Text variant="highlight">Highlight</Text>
-                    </Box>
-                    <Box>
-                        <Text variant="body">Body</Text>
-                    </Box>
-                    <Box>
-                        <Text variant="callout">Callout</Text>
-                    </Box>
-                    <Box>
-                        <Text variant="hint">Hint</Text>
-                    </Box>
-                    <Box>
-                        <Text variant="label">Label</Text>
-                    </Box>
+                    <BottomModal
+                        isVisible={isModalVisible}
+                        onVisibilityChange={setIsModalVisible}
+                        title="Typography Demo"
+                        hasBackArrow
+                        onBackArrowClick={() => setIsModalVisible(!isModalVisible)}
+                        modalTrigger={
```

Comment:
we don't need `modalTrigger` at all, just place button anywhere and it will still work so you can get rid of this prop

Reply:
Yeah I thought about that as well but to me it seems like this way it's very clear which element triggers the modal. For example it happens sometimes that the modal could be way below the actual modal and this will force-group these things together. I don't think it brings any benefit to not accept this prop but there is some readability / developer experience benefit with this prop imho.

Reply:
I think the oposite :D Using props that are not necessary is never good practise and forcing to have modal together with trigger si also not good practise, lot of this you just want to keep modals at the end of render or in completely different file, but you want to have trigger in origin file because it's form input and you need to pass some props here and passing through another component is unecessary.

You can check other libraries and they are doing it this way:
https://mui.com/material-ui/react-modal/
https://callstack.github.io/react-native-paper/modal.html

Reply:
Yes, the same structure is achieved within the children component which I think is reasonable because you usually want to render the trigger in same section as the modal. But I will decouple them if you think it's more flexible.

Reply:
[7a9b5b8](https://github.com/trezor/trezor-suite/pull/5493/commits/7a9b5b818c0d5cf81b0c93ae7f749584b99cadf7)

Reply:
They are using children for modal content not trigger